### PR TITLE
Fix email link on iOS Safari in amp-social-share

### LINKED ISSUE #3805: Amp-social-share doesn't open email agent on IOS
When using amp-social-share the email agent is not being opened on IPhone and IPOD 6 using Safari browser.

## How do we reproduce the issue?
1. Open the safari browser in an Iphone 5 or 6.
2. Open the AMP page http://app.eltiempo.com/contenido/politica/partidos-politicos/ARTICULO-WEB-AMP-16624792.html
3. Click in the email icon between gplus and whatsapp..
4. It will open a blank screen and no email agents.

## What browsers are affected?
Safari

All browsers? Some specific browser? What device type?
IOS in an IPhone 5 and 6

## Which AMP version is affected?
Powered by AMP ⚡ HTML – Version 1466794223483
<script async src="https://cdn.ampproject.org/v0.js"></script>
<script async custom-element="amp-social-share" src="https://cdn.ampproject.org/v0/amp-social-share-0.1.js"></script>

Is this a new issue? Or was it always broken? Paste your AMP version. You can find it in the browser dev tools.
Always broken.
Powered by AMP ⚡ HTML – Version 1466794223483

### PR DESCRIPTION
**Pull request by mkhatib**
Fixes #3805

### CODE REVIEW

File: extensions/amp-social-share/0.1/amp-social-share.js [hunk 1]
```diff
@@ -49,12 +50,24 @@ class AmpSocialShare extends AMP.BaseElement {
...
+
+    /** @private {string} */
+    this.target_ = null;
+
     const hrefWithVars = addParamsToUrl(this.shareEndpoint_, this.params_);
     const urlReplacements = urlReplacementsFor(this.win);
     urlReplacements.expand(hrefWithVars).then(href => {
       this.href_ = href;
+      // mailto: protocol breaks when opened in _blank on iOS Safari.
+      const tgtLoc = parseUrl(href);
```
Comment (lannka) [inline extensions/amp-social-share/0.1/amp-social-share.js]: nit: can be inlined

File: extensions/amp-social-share/0.1/test/test-amp-social-share.js [hunk 2]
```diff
@@ -166,4 +175,18 @@ describe('amp-social-share', () => {
       )).to.be.true;
     });
   });
+
+  it('opens mailto: window in _self on iOS Safari', () => {
+    isIos = true;
+    isSafari = true;
+    return getShare('email').then(el => {
+      el.implementation_.handleClick_();
+      expect(el.implementation_.win.open.called).to.be.true;
```
Comment (lannka) [inline extensions/amp-social-share/0.1/test/test-amp-social-share.js]: `expect(xx).to.be.calledOnce()`

File: extensions/amp-social-share/0.1/test/test-amp-social-share.js [hunk 3]
```diff
@@ -166,4 +175,18 @@ describe('amp-social-share', () => {
...
+      expect(el.implementation_.win.open.calledWith(
```
Comment (lannka) [inline extensions/amp-social-share/0.1/test/test-amp-social-share.js]: `expect(..).to.be.calledWith(..)`

**Comment by lannka** [comment]
LGTM with some minor comments.

File: extensions/amp-social-share/0.1/amp-social-share.js (hunk 1 above)
Comment (mkhatib) [inline extensions/amp-social-share/0.1/amp-social-share.js]: Done

File: extensions/amp-social-share/0.1/test/test-amp-social-share.js (hunk 2 above)
Comment (mkhatib) [inline extensions/amp-social-share/0.1/test/test-amp-social-share.js]: Done

File: extensions/amp-social-share/0.1/test/test-amp-social-share.js (hunk 3 above)
Comment (mkhatib) [inline extensions/amp-social-share/0.1/test/test-amp-social-share.js]: Done

**Pull request merged**